pydoctor/sphinx_ext/build_apidocs.py

```diff
@@ -76,14 +89,55 @@ def _run_pydoctor(name: str, arguments: Iterable[str], placeholders: Mapping[str
             logger.warning(line)
 
 
+def update_intersphinx_mapping(app: Sphinx, config: Config) -> None:
+    """
+    Called to build and inject our own intersphinx inventory object.
+    """
+    rtd_version = 'latest'
+    if os.environ.get('READTHEDOCS', '') == 'True':
+        rtd_version = os.environ.get('READTHEDOCS_VERSION', 'latest')
+
+    output_dir = pathlib.Path(app.outdir)
+
+    intersphinx_mapping = config.intersphinx_mapping
+    for key, value in config.pydoctor_intersphinx_mapping.items():
+        url, target_path = value
+
+        url = url.format(**{'rtd_version': rtd_version})
+        inventory_path = output_dir / target_path
+
+        intersphinx_mapping[key] = (url, str(inventory_path))
+
+    if not config.pydoctor_args:
+        raise ConfigError("Missing 'pydoctor_args'.")
```

Comment:
It would be cleaner to perform input validation before making any state changes.

Reply:
True. This was updated.

But the build was existing anyway.

Reply:
Yeah, it might not matter here, it's just a general good habit.